src/main/java/io/cryostat/credentials/CredentialCheck.java

```diff
@@ -42,7 +40,7 @@ public class CredentialCheck {
     @Blocking
     @RolesAllowed("read")
     @Path("/api/beta/credentials/{connectUrl}")
-    public Uni<V2Response> checkCredentialForTarget(
+    public Uni<CredentialTestResult> checkCredentialForTarget(
```

Comment:
@aali309 take a look at this change please - in this case it's better to just return the result type directly, rather than wrapping it in a JAX-RS Response. Quarkus will automatically set the Content-Type response header anyway, so this saves a couple of lines of code but also makes sure that the content negotiation between the Cryostat server and the user's browser properly determines the response format and header value.

Reply:
This same simplification should be applied elsewhere too, unless there is a particular reason to wrap the result in a Response, but I think most of the time there won't be.